sdcm/teardown_validators/events.py

```diff
@@ -0,0 +1,37 @@
+import json
+import logging
+
+from sdcm.sct_events.events_device import get_events_main_device
+from sdcm.sct_events import Severity
+from sdcm.teardown_validators.base import TeardownValidator
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ErrorEventsValidator(TeardownValidator):  # pylint: disable=too-few-public-methods
+    """
+    A teardown validator that checks test events log for presence of specific error events and critical-level events.
+    If the test doesn't have any of the defined failing error events or critical events, its status
+    defaults to 'SUCCESS'.
+
+    Usage:
+        An instance of ErrorEventsValidator can be used during teardown of a test, where specific error events
+        should not be considered as test failures.
+        This is particularly useful in complex testing environments where multiple parallel operations may lead to
+        non-critical errors that should not directly influence the overall test outcome.
+    """
+    validator_name = 'test_error_events'
+
+    def validate(self):
+        LOGGER.info("Running validation of ERROR events for parallel nemeses")
+        failing_event_types = [et.lower() for et in self.configuration.get("failing_events")]
```

Comment:
are you sure it works when lowering event name? 

Would be good to have it unit tested.

Reply:
Right. Regex search approach was performed with ignorecase modifier, but after switching to parsing events to json we need to take care of this ourselves.
Fixed this and added `unit_tests/test_teardown_validator.py` unit tests. Their results are in https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-7651/4/